pkg/util/net/ip/ip.go

```diff
@@ -82,3 +85,28 @@ func isIPv6Disabled(filename string) bool {
 
 	return data[0] == '1'
 }
+
+func ChangeCIDRsToIPs(ifs *api.InterfaceStatus) error {
+	// if IP address was the passed value we should do nothing
+	if net.ParseIP(ifs.Ip) == nil {
+		ip, _, err := net.ParseCIDR(ifs.Ip)
+		if err != nil {
+			return fmt.Errorf("qemu-guest-agent reported invalid CIDR value '%s' for '%s': %v",
+				ifs.Ip, ifs.InterfaceName, err)
+		}
+		ifs.Ip = ip.String()
```

Comment:
Making this change mutable is not safe as it may change the values even if failure occurs later on, causing recovery to be questionable.
Try using a mutable implementation, returninig the data so the caller can update it (or not, in case of error).